src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -81,6 +81,33 @@ def _make_causal_mask(input_ids_shape: torch.Size, dtype: torch.dtype, past_key_
         mask = torch.cat([torch.zeros(tgt_len, past_key_values_length, dtype=dtype), mask], dim=-1)
     return mask[None, None, :, :].expand(bsz, 1, tgt_len, tgt_len + past_key_values_length)
 
+# To only condition from certain indexes
+def _make_variable_start_mask(input_ids_shape: torch.Size, dtype: torch.dtype, per_sample_start: torch.tensor,
```

Comment:
Do you think we could just define one function `_make_causal_mask` which takes `per_sample_start` as an optional argument? 

If this arg defaults to `None`, then we due our usual causal mask function. 

If it's not `None` and `per_sample_start.sum() > 1`, then we only condition on certain indices using this modified algorithm?

I think this way we repeat less of the code and make the control flow a bit easier to understand. WDYT?

Reply:
makes sense!